tools/render-test/render-d3d11.cpp

```diff
@@ -356,19 +366,19 @@ SlangResult D3D11Renderer::initialize(void* inWindowHandle)
     for (int ii = 0; ii < 2; ++ii)
     {
         const HRESULT hr = D3D11CreateDeviceAndSwapChain_(
-            NULL,                    // adapter (use default)
+            nullptr,                    // adapter (use default)
             D3D_DRIVER_TYPE_REFERENCE,
             //D3D_DRIVER_TYPE_HARDWARE,
-            NULL,                    // software
+			nullptr,                    // software
             deviceFlags,
             &featureLevels[ii],
             totalNumFeatureLevels - ii,
             D3D11_SDK_VERSION,
             &swapChainDesc,
-            &m_swapChain,
-            &m_device,
+            m_swapChain.writeRef(),
```

Comment:
This is as good a place as any to make sure I understand the `readRef()` and `writeRef()` methods. Am I correct (without double-checking the implementation) that:

- `readRef()` just returns what amounts to `(T**) this`. That is, the address of the underlying raw pointer as a raw pointer-pointer. It might be `T* const*` to avoid people writing to it.

- `writeRef()` needs to release and null out any value currently pointed to, and then return `(T**) this` so that it can be written to. It assumes that the write will be an "attach" operation, so that reference counts don't need to be adjusted on the write side.